py/runtime0.h

```diff
@@ -76,6 +76,12 @@ typedef enum {
     MP_UNARY_OP_HASH, // __hash__; must return a small int
     MP_UNARY_OP_ABS, // __abs__
     MP_UNARY_OP_INT, // __int__
+    #if MICROPY_PY_BUILTINS_FLOAT
+    MP_UNARY_OP_FLOAT, // __float__
+    #endif
+    #if MICROPY_PY_BUILTINS_COMPLEX
```

Comment:
same as above, no need to have this #if-guard

Reply:
OK. I'll change that.

Reply:
Actually it might add to the code size because it changes the number of runtime special methods and over in `objtype.c` you have an array of (small, one-bytel) QSTR numbers for the names of the special methods. Depending on the alignment rules for the array it could add a couple of bytes.

Reply:
> Actually it might add to the code size because it changes the number of runtime special methods

Yes you're right.  But I'd be happy to pay a few bytes to keep this clean, without #if's.  Already `MP_UNARY_OP_SIZEOF` will add 1 byte even if not used.

Reply:
In fact it's worse, since we also either end up adding the QSTRs for `__float__` and `__complex__` or we need to add a check if the name is present. It looks like such a check was once present in the past so I'll use that.